### PR TITLE
[Core] Support ConcurrentGroup part1

### PR DESCRIPTION
**Pull request by jovany-wang**
## Why are these changes needed?
This PR includes the core worker changes and Java API of concurrency group.

## Related issue number
#16047

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by jovany-wang** [comment]
@wuisawesome @simon-mo @ericl 
Do you have bandwidths to take a review?

File: src/ray/core_worker/transport/direct_actor_transport.cc
Comment (WangTaoTheTonic) [inline src/ray/core_worker/transport/direct_actor_transport.cc:510]: This statement is not necessary. The variable could be passed into constructor directly.

File: src/ray/core_worker/transport/direct_actor_transport.cc
Comment (WangTaoTheTonic) [inline src/ray/core_worker/transport/direct_actor_transport.cc:559]: `max_concurrency_ ` is number of threads in default thread pool, maybe inside `ActorSchedulingQueue ` we can rename it to `default_ max_concurrency_ ` or something?

Comment (WangTaoTheTonic) [reply src/ray/core_worker/transport/direct_actor_transport.cc:559]: After recheck, this parameter is only used for create `fiber_state`, we can rename it to fiber_thread_num or something.

File: src/ray/protobuf/common.proto
Comment (WangTaoTheTonic) [inline src/ray/protobuf/common.proto:212]: Clear this TODO :)

**Review by WangTaoTheTonic — APPROVED** [review]
LGTM just some style and tiny comments.
BTW don't forget to change `Concurrent` to `Concurrency` in PR title.

**Comment by jovany-wang** [comment]
Hi @ericl , do you guys want to take a look at this PR? This PR includes Java API change and core change, not include Python API change. If you guys have no bandwidth to review, we're going to merge this once there're some approvals from our side. Thanks.

**Comment by ericl** [comment]
Sure. I think we're fine with the high level API, will take a look

File: java/api/src/main/java/io/ray/api/call/ActorCreator.java
Comment (ericl) [reply java/api/src/main/java/io/ray/api/call/ActorCreator.java:48]: +1

**Review by ericl — APPROVED** [review]

File: src/ray/core_worker/transport/direct_actor_transport.cc
Comment (jovany-wang) [reply src/ray/core_worker/transport/direct_actor_transport.cc]: It's not thread number, it's the concurrency number for fiber mode in one thread. I rename it to `fiber_max_concurrency` instead.

**Comment by jovany-wang** [comment]
@WangTaoTheTonic @ericl  
Thanks for the reviewing. All CI passed, and I'm merging this PR.

**Pull request merged**
